exist-core/src/main/antlr/org/exist/xquery/parser/XQuery.g

```diff
@@ -103,6 +103,10 @@ options {
 		return buf.toString();
 	}
 
+	public Exception getLastException() {
+        return (Exception) exceptions.get(exceptions.size() - 1);
```

Comment:
size can't return 0?

Reply:
It is actually only every called if `XQueryParser#foundErrors()` returns true first, however just to make the intention clear I added and additional check there to avoid a potential `IndexOutOfBoundsException`